sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -0,0 +1,735 @@
+
+                // properly format binding types by removing "Attribute" and "Input" descriptors
+                string bindingType = attributeName.TrimStringsFromEnd(new string[] { "Attribute", "Input", "Output" });
+
+                // Set binding direction
+                string bindingDirection = SymbolEqualityComparer.Default.Equals(bindingAttrData.AttributeClass?.BaseType, Compilation.GetTypeByMetadataName(Constants.OutputBindingAttributeType)) ? "Out" : "In";
+
+                bindings = new Dictionary<string, string>();
+                bindings.Add("name", FormatObject(bindingName));
+                bindings.Add("type", FormatObject(bindingType));
+                bindings.Add("direction", FormatObject(bindingDirection));
+
+                // Add additional bindingInfo to the anonymous type because some functions have more properties than others
+                foreach (var prop in attributeProperties!) // attributeProperties won't be null here b/c we would've exited this method earlier if it was during TryGetAttributeProperties check
+                {
+                    var propertyName = prop.Key;
+
+                    if (prop.Value.GetType().IsArray)
+                    {
+                        string arr = FormatArray((IEnumerable)prop.Value);
+                        bindings[propertyName] = arr;
+                    }
+                    else
+                    {
+                        var isEnum = false;
+                        
+                        if (string.Equals(prop.Key.ToString(), "authLevel", StringComparison.OrdinalIgnoreCase)) // prop keys come from Azure Functions defined attributes so we can check directly for authLevel
+                        {
+                            isEnum = true;
+                        }
+
+                        var propertyValue = FormatObject(prop.Value, isEnum);
+                        bindings[propertyName] = propertyValue;
+                    }
+                }
+
+                return true;
+            }
+
+            private bool TryGetAttributeProperties(IMethodSymbol attribMethodSymbol, AttributeData attributeData, Location attribLocation, out IDictionary<string, object>? attrProperties)
+            {
+                attrProperties = new Dictionary<string, object>();
+
+                if (attributeData.ConstructorArguments.Any())
+                {
+                    if (!TryLoadConstructorArguments(attribMethodSymbol, attributeData, attrProperties, attribLocation))
+                    {
+                        attrProperties = null;
+                        return false;
+                    }
+                }
+
+                foreach (var namedArgument in attributeData.NamedArguments)
+                {
+                    if (namedArgument.Value.Value != null)
+                    {
+                        if (String.Equals(namedArgument.Key, Constants.IsBatchedKey) && !attrProperties.Keys.Contains("Cardinality"))
```

Comment:
The json generator does not, and I think we don't have to worry because this is something that comes from our defined attributes (it's not something that the user defines/adds as add-on properties)